test/actions.test.js

```diff
@@ -35,34 +35,31 @@ test('the `post` action creator allows different bodies and headers', () => {
   );
 });
 
-test('the `post.json` action creator is correct', () => {
+test('the `del` action creator is correct', () => {
   const success = jest.fn();
   const fail = jest.fn();
-  const body = { hello: 'world' };
 
   const expected = {
     type: FETCH,
-    payload: {
-      method: 'POST',
-      url: '/v1',
-      body: JSON.stringify(body),
-      success,
-      fail,
-      headers: { 'Content-Type': 'application/json' }
-    },
+    payload: { method: 'DELETE', url: '/v1', success, fail },
     error: null
   };
-  expect(post.json('/v1', { body, success, fail })).toEqual(expected);
+  expect(del('/v1', { success, fail })).toEqual(expected);
 });
 
-test('the `del` action creator is correct', () => {
+test('the `.json` action creators auto-convert json', () => {
   const success = jest.fn();
-  const fail = jest.fn();
 
-  const expected = {
-    type: FETCH,
-    payload: { method: 'DELETE', url: '/v1', success, fail },
-    error: null
-  };
-  expect(del('/v1', { success, fail })).toEqual(expected);
+  const body = { hello: 'world' };
+  const mockResponseData = { status: 'ok' };
+  const mockResponseObj = new Response(JSON.stringify(mockResponseData));
+
+  const action = post.json('/v1', { body, success });
+  const payload = action.payload;
+  expect(payload.body).toEqual(JSON.stringify(body));
+  expect(payload.headers).toEqual({ 'Content-Type': 'application/json' });
+
+  return payload
```

Comment:
can we use `async/await` in this maybe?

Reply:
i tried initially and it didn't like me using `await` in tests: `/Users/francoiscampbell/Software/redux-saga-fetch/test/actions.test.js: await is a reserved word (63:20)`

Reply:
yeah we have to configure `babel-jest` i think